aiida_mlip/parsers.py

```diff
@@ -1,58 +1,136 @@
+    ------
+    exceptions.ParsingError
+        If the ProcessNode being passed was not produced by a Singlepointcalc.
     """
 
     def __init__(self, node):
         """
-        Initialize Parser instance
-
-        Checks that the ProcessNode being passed was produced by a DiffCalculation.
+        Check that the ProcessNode being passed was produced by a Singlepointcalc.
 
-        :param node: ProcessNode of calculation
-        :param type node: :class:`aiida.orm.nodes.process.process.ProcessNode`
+        Parameters
+        ----------
+        node : aiida.orm.nodes.process.process.ProcessNode
+            ProcessNode of calculation.
         """
         super().__init__(node)
-        if not issubclass(node.process_class, DiffCalculation):
-            raise exceptions.ParsingError("Can only parse DiffCalculation")
+        if not issubclass(node.process_class, Singlepointcalc):
+            raise exceptions.ParsingError("Can only parse Singlepointcalc")
 
+    # pylint: disable=too-many-locals
     def parse(self, **kwargs):
         """
-        Parse outputs, store results in database.
+        Parse outputs, store results in the database.
+
+        Parameters
+        ----------
+        **kwargs : Any
+            Any keyword argument.
 
-        :returns: an exit code, if parsing fails (or nothing if parsing succeeds)
+        Returns
+        -------
+        int
+            An exit code.
         """
         output_filename = self.node.get_option("output_filename")
+        xyzoutput = (self.node.inputs.xyz_output_name).value
+        logoutput = (self.node.inputs.log_filename).value
+
+        remote_folder = self.node.get_remote_workdir()
 
         # Check that folder content is as expected
         files_retrieved = self.retrieved.list_object_names()
-        files_expected = [output_filename]
+
+        files_expected = [xyzoutput, logoutput]
         # Note: set(A) <= set(B) checks whether A is a subset of B
         if not set(files_expected) <= set(files_retrieved):
             self.logger.error(
                 f"Found files '{files_retrieved}', expected to find '{files_expected}'"
             )
             return self.exit_codes.ERROR_MISSING_OUTPUT_FILES
 
-        # add output file
-        self.logger.info(f"Parsing '{output_filename}'")
+        # Add output file to the outpus
+        self.logger.info(f"Parsing '{xyzoutput}'")
+
+        with self.retrieved.open(logoutput, "rb") as handle:
+            log_node = SinglefileData(file=handle)
+
+        self.out("log_output", log_node)
+
+        with self.retrieved.open(xyzoutput, "rb") as handle:
+            xyz_node = SinglefileData(file=handle)
+
+        self.out("xyz_output", xyz_node)
+
         with self.retrieved.open(output_filename, "rb") as handle:
-            output_node = SinglefileData(file=handle)
-        self.out("mlip", output_node)
+            stdout_node = SinglefileData(file=handle)
+
+        self.out("std_output", stdout_node)
```

Comment:
Would this work?
```suggestion
        with self.retrieved.open(logoutput, "rb") as handle:
            self.out("log_output", SinglefileData(file=handle))

        with self.retrieved.open(xyzoutput, "rb") as handle:
            self.out("xyz_output", SinglefileData(file=handle))

        with self.retrieved.open(output_filename, "rb") as handle:
            self.out("std_output",  SinglefileData(file=handle))
```
With 3.9+ (parens on with) you could do this as:
```suggestion
        with (self.retrieved.open(logoutput, "rb") as log_handle,
              self.retrieved.open(xyzoutput, "rb") as xyz_handle,
              self.retrieved.open(output_filename, "rb") as out_handle):
            self.out("log_output", SinglefileData(file=log_handle))
            self.out("xyz_output", SinglefileData(file=xyz_handle))
            self.out("std_output",  SinglefileData(file=out_handle))
```